tests/drivers/candev/main.c

```diff
@@ -221,34 +307,44 @@ if (IS_ACTIVE(CONFIG_USE_LOOPBACK_MODE)) {
     candev->driver->set(candev, CANOPT_STATE, &mode, sizeof(mode));
 }
 
+/* Depending from the CAN controller used, this test example will provide different results.
+For MCP2515 standalone CAN controller, the last filter won't be applied as the first reception mailbox supports up to two filters
+For SAMD5x/E5x CAN controller, and with keeping the default parameters in candev_samd5x.h, the last filter won't be applied
+    as the CAN controller supports up to 3 standard filters
+For SAMD5x/E5x CAN controller, if you increase the maximum capacity of the standard filters (as mentioned in Makefile.board.dep),
+    the last filter can be applied correctly. */
+#if defined(MODULE_MCP2515)
     if (IS_ACTIVE(MCP2515_RECV_FILTER_EN)) {
+#endif
         /* CAN filters examples */
         struct can_filter filter[4];
         filter[0].can_mask = 0x7FF;
         filter[0].can_id = 0x001;
-#if defined(MODULE_MCP2515)
-        filter[0].target_mailbox = 0;   /* messages with CAN ID 0x001 will be received in mailbox 0 */
+#if !defined(__linux__)
```

Comment:
this is still wrong, it is not a linux thing to not provide maliboxes

Reply:
Wrong? how? CAN on native is not supporting the mailboxes

Reply:
but other cans aren't mailboxed as well -> mailbox is a feature that not every can provides and being linux is a bad indicator for not having that feature. 

if we leave it  there will be a separate PR to fix this

Reply:
AFAIK all the CAN controllers provide at least two mailboxes (if you know of any other case, please share it)

Reply:
atmega16/32/64m - at90can128

Reply:
On a second thought, what is the problem of having `target_mailbox` even though there is just one mailbox? That will require a little change in the native`s `can.c`

Reply:
why not get things right?

Reply:
It's not right to have a mailbox for the native board?

Reply:
yes it is not it is also not right to have mailbox for at90can128

Reply:
What?!
![image](https://github.com/RIOT-OS/RIOT/assets/53952217/ee5d903d-4015-47cb-a8f5-9ffc93eca0d6)
Do you see what's called in the datasheet? A **mailbox**, it's a reception place for the CAN controller, but it just has one. The driver then needs to handle that, exactly like in this driver

Reply:
a misnamed buffer isn't a mailbox feature that we need to care about

Reply:
no problem i will do a cleanup PR

Reply:
> > a misnamed buffer isn't a mailbox feature that we need to care about
> 
> what? misnamed? No the mailbox itself contains data buffers (seen as FIFO/queue, depending from configuration).

there is one "mailbox" <- this is the way it is written in the at90can manual, we do not need to care about that in the driver since you wont choose another one. 
![image](https://github.com/RIOT-OS/RIOT/assets/814189/3594ea83-4d95-4a63-9a6c-c3709117fc88)

It contains MOb which will filter and buffer one message. (one filter per MOb) 

> 
> > no problem i will do a cleanup PR
> 
> I'm trying to find a solution middle way here with your review. I'm offering that we include the mailboxes for every board as it's a component of every CAN controller (You'll find this term in every CAN controller datasheet). Additionally this can provide to extend the CAN driver in the native board with another mailbox if it's possible.

Mailboxes are a hardware feature they are either there (more than one so we need to care) or not, and if they are there then they might be exposed or not.

Reply:
If we have just one mailbox, then the driver will make the check if that one mailbox number is selected (e.g `filter.target_mailbox=0`), if not fail an assertion. And reception/transmission mailboxes are available for every CAN controller, just the number of mailboxes and their capacity change

Reply:
a new assertion isn't the answer 

as you say mailboxes are a common feature but their number isn't changing very it is either don't care there is just one or there are two